### PR TITLE
Remove the blocklist for `aten.div.Tensor`

### LINKED ISSUE #479: aten.div.Tensor

### PR DESCRIPTION
**Pull request by jdh8**
### Ticket
- Resolves #479 
- Relates to tenstorrent/tt-metal#15630

### Problem description
Remove the blocklist for `aten.div.Tensor` and make sure these cases pass

### What's changed
- [x] Further optimize `tensor / const` to `tensor * (1 / const)`
- [x] Remove block list entries that should work since #507
- [ ] Resolve `tensor / scalar` where the scalar is unknown

### CODE REVIEW

**Comment by swimdi** [comment]
This PR seems no remove the blocklist?

**Comment by jdh8** [comment]
I cannot remove the block list yet.  It still does not work when the denominator is an unknown scalar.  This is the main reason why this PR is still a draft.

**Comment by swimdi** [comment]
okok, thanks for remind

**Comment by jdh8** [comment]
I'm marking this PR as ready.  I remove block list entries that should work since #507.  The failing cases are now tracked by tenstorrent/tt-metal#15630.  I also make them `xfail` test cases.

**Comment by jdh8** [comment]
@swimdi, how do we regenerate the block list for status quo?  Some worrisome test cases seem to fall through.

**Comment by swimdi** [comment]
You can
1. Disable all the blocklist
2. `pytest tests/autogen_op/ALL` to get which cases are failed
3. `python3 tools/generate_guard_function_from_input_var_metric.py`
     - @jerrysky3 is added some conv OOM case on to_tt_guard_autogen, so need to add it on the new one

File: torch_ttnn/passes/lowering/to_tt_pass.py [hunk 1]
```diff
@@ -601,10 +601,14 @@ def batch_norm_inference(input, weight, bias, mean, var, momentum, eps):
                 return g.call_function(ttnn.log, (softmax_node,), kwargs)
 
             if node.target == torch.ops.aten.div.Tensor:
-                if not isinstance(args[1], (float, int)) and (get_shape(args[0]) != get_shape(args[1])):
-                    recip = g.call_function(ttnn.reciprocal, (args[1],), {})
-                    return g.call_function(ttnn.mul, (args[0], recip), {})
-                return g.call_function(ttnn.div, args, {})
+                if isinstance(args[1], (float, int)):
+                    return g.call_function(ttnn.mul, (args[0], 1.0 / args[1]), {})
```
Comment (jdh8) [inline torch_ttnn/passes/lowering/to_tt_pass.py:605]: Done!  I checked the additional cases.  Some constant denominators are `int`.  I handle them here as well.

**Review by jerrysky3 — APPROVED** [review]

**Review by swimdi — APPROVED** [review]

**Comment by jdh8** [comment]
Missing dependency?
```
/home/ubuntu/actions-runner/_work/_tool/Python/3.8.18/x64/lib/python3.8/site-packages/_pytest/config/__init__.py:329: PluggyTeardownRaisedWarning: A plugin raised an exception during an old-style hookwrapper teardown.
Plugin: helpconfig, Hook: pytest_cmdline_parse
ConftestImportFailure: ImportError: libGL.so.1: cannot open shared object file: No such file or directory (from /home/ubuntu/actions-runner/_work/pytorch2.0_ttnn/pytorch2.0_ttnn/tests/conftest.py)
For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
  config = pluginmanager.hook.pytest_cmdline_parse(
ImportError while loading conftest '/home/ubuntu/actions-runner/_work/pytorch2.0_ttnn/pytorch2.0_ttnn/tests/conftest.py'.
tests/conftest.py:6: in <module>
    from tests.utils import calculate_accuracy
../../_tool/Python/3.8.18/x64/lib/python3.8/site-packages/tests/__init__.py:3: in <module>
    from ultralytics.utils import ASSETS, ROOT, WEIGHTS_DIR, checks, is_dir_writeable
../../_tool/Python/3.8.18/x64/lib/python3.8/site-packages/ultralytics/__init__.py:11: in <module>
    from ultralytics.data.explorer.explorer import Explorer
../../_tool/Python/3.8.18/x64/lib/python3.8/site-packages/ultralytics/data/__init__.py:3: in <module>
    from .base import BaseDataset
../../_tool/Python/3.8.18/x64/lib/python3.8/site-packages/ultralytics/data/base.py:12: in <module>
    import cv2
../../_tool/Python/3.8.18/x64/lib/python3.8/site-packages/cv2/__init__.py:181: in <module>
    bootstrap()
../../_tool/Python/3.8.18/x64/lib/python3.8/site-packages/cv2/__init__.py:153: in bootstrap
    native_module = importlib.import_module("cv2")
E   ImportError: libGL.so.1: cannot open shared object file: No such file or directory
Failure: Tests failed with exit code 4.
```

**Pull request merged**